pairinteraction_gui/pairinteraction/app.py

```diff
@@ -2578,486 +2594,505 @@ def forbidSamebasis(self):
                 self.ui.checkbox_system_samebasis.setEnabled(True)
                 self.ui.checkbox_system_samebasis.setCheckState(self.samebasis_state)
 
+    @QtCore.pyqtSlot()
+    def defaultQuantumnumbers(self):
+        sender = self.sender()
+        number = int(sender.objectName()[-1])
+        species = getattr(self.ui, f"combobox_system_species{number}").currentText()
+        s = SPIN_DICT.get(species, 0.5)
+
+        for q in ["j", "m"]:
+            for system in ["system", "plot"]:
+                spinbox = getattr(self.ui, f"spinbox_{system}_{q}{number}")
+                value = spinbox.value()
+                spinbox.setDecimals(0 if s % 1 == 0 else 1)
+                # Minimum and Maximum for m are handled via setDecimals
+                if spinbox.minimum() >= 0:
+                    spinbox.setMinimum(s % 1)
+                spinbox.setValue(np.floor(value) + (s % 1))
+
     @QtCore.pyqtSlot()
     def validateCores(self):
         sender = self.sender()
         if sender.value() != -1 and sender.value() < 2:
             sender.setValue(2)
 
-    @QtCore.pyqtSlot()
-    def validateQuantumnumbers(self):
+    def validateAllQuantumnumbers(self):
```

Comment:
The check allows for "Sr3, l=0, and j=0", although it must be j=1 if l=0 for the triplet state.